README.md

```diff
@@ -24,88 +22,117 @@ Via CDN:
+
+When you login a user, you can request scopes up front if you know they will be needed, but it is not required, and login can be called with no params. It is best practice to only request scopes you need when you need them, a concept called dynamic consent. AAD will only allow you to get consent for 3 resources at a time, although you can request many scopes within a resource.
 
 ```JavaScript
-   var graphScopes = ["user.read", "mail.send"];
+   var loginRequest = {
+       scopes: ["user.read", "mail.send"] // optional Array<string>
+   };
 
-   userAgentApplication.loginPopup(graphScopes).then(function (idToken) {
-       //login success
-   }, function (error) {
-       //login failure
-       console.log(error);
-   });
+    msalInstance.loginPopup(loginRequest)
+        .then(response => {
+            // handle response
+        })
+        .catch(err => {
+            // handle error
+        });
 
 ```
-> Note: The scopes passed to the login method are optional. In this example, the graphScopes are passed in the login method to obtain consent upfront from the user for your app to access certain Graph API scopes. The idtoken returned here does not contain the scopes. In the next step, you can see how to get an access token which will contain the consented scopes.
 
 #### 3. Get an access token to call an API
 
-In MSAL, you can get access tokens for the APIs your app needs to call using the acquireTokenSilent method which makes a silent request(without prompting the user) to Azure AD to obtain an access token. The Azure AD service then returns an [access token](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens) containing the user consented scopes to allow your app to securely call the API.  
+In MSAL, you can get access tokens for the APIs your app needs to call using the `acquireTokenSilent` method which makes a silent request(without prompting the user with UI) to Azure AD to obtain an access token. The Azure AD service then returns an [access token](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens) containing the user consented scopes to allow your app to securely call the API.
 
-If the silent token acquisition fails for some reasons such as an expired token or password change, you will need to invoke an interactive method to acquire tokens such as acquireTokenPopup or acquireTokenRedirect.
+If the `acquireTokenSilent` call fails with an error of type `InteractionRequiredAuthError` (requiring the user to consent to something), you will need to initiate an interactive request. You can use `acquireTokenRedirect` or `acquireTokenPopup`.
 
- ```JavaScript
-    var graphScopes = ["user.read", "mail.send"];
+`acquireTokenSilent` will look for a valid token in the cache, and if it is close to expiring or does not exist, will automatically refresh it for you.
```

Comment:
@navyasric can you confirm if any more data is needed here regarding the 'tokenRenewalOffsetSeconds'?